app/src/sharedTest/java/org/oppia/app/profile/AdminPinActivityTest.kt

```diff
@@ -110,6 +112,86 @@ class AdminPinActivityTest {
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_confirm_pin)))).perform(scrollTo(),typeText("1234"), closeSoftKeyboard())
+      onView(withId(R.id.submit_button)).perform(scrollTo(),click())
+      onView(allOf(withId(R.id.error_text), isDescendantOfA(withId(R.id.input_confirm_pin)))).check(matches(withText(context.getString(R.string.admin_pin_error_pin_confirm_wrong))))
+    }
+  }
+
+  @Test
+  fun testAdminPinActivity_rotateScreen_inputPin_inputWrongConfirmPin_clickSubmit_inputConfirmPin_checkErrorIsCleared() {
+    ActivityScenario.launch<AdminPinActivity>(AdminPinActivity.createAdminPinActivityIntent(context, 0, -10710042)).use {
+      onView(ViewMatchers.isRoot()).perform(OrientationChangeAction.orientationLandscape())
+      closeSoftKeyboard()
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_pin)))).perform(typeText("12345"), closeSoftKeyboard())
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_confirm_pin)))).perform(scrollTo(),typeText("1234"), closeSoftKeyboard())
+      onView(withId(R.id.submit_button)).perform(scrollTo(),click())
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_confirm_pin)))).perform(scrollTo(),typeText("5"), closeSoftKeyboard())
+      onView(allOf(withId(R.id.error_text), isDescendantOfA(withId(R.id.input_confirm_pin)))).check(matches(withText("")))
+    }
+  }
+
+  @Test
+  fun testAdminPinActivity_inputPin_inputWrongConfirmPin_clickSubmit_rotateScreen_checkConfirmWrongError(){
+    ActivityScenario.launch<AdminPinActivity>(AdminPinActivity.createAdminPinActivityIntent(context,0,-10710042)).use {
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_pin)))).perform(typeText("12345"), closeSoftKeyboard())
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_confirm_pin)))).perform(scrollTo(),typeText("54321"), closeSoftKeyboard())
+      onView(withId(R.id.submit_button)).perform(click())
+      onView(ViewMatchers.isRoot()).perform(OrientationChangeAction.orientationLandscape())
+      onView(allOf(withId(R.id.error_text), isDescendantOfA(withId(R.id.input_confirm_pin)))).check(matches(withText(R.string.admin_pin_error_pin_confirm_wrong)))
+    }
+  }
+
+  @Test
+  fun testAdminPinActivity_inputShortPin_clickSubmit_rotateScreen_checkPinLengthError() {
+    ActivityScenario.launch<AdminPinActivity>(AdminPinActivity.createAdminPinActivityIntent(context, 0, -10710042)).use {
+      onView(allOf(withId(R.id.input), isDescendantOfA(withId(R.id.input_pin)))).perform(typeText("123"), closeSoftKeyboard())
+      onView(withId(R.id.submit_button)).perform(scrollTo(), click())
+      onView(ViewMatchers.isRoot()).perform(OrientationChangeAction.orientationLandscape())
+      onView(allOf(withId(R.id.error_text), isDescendantOfA(withId(R.id.input_pin)))).check(matches(withText(R.string.admin_pin_error_pin_length)))
+    }
+  }
+
+
   private fun setUpTestApplicationComponent() {
```

Comment:
Once you fix all changes in this file, please check that there are no unused imports.